Create Git Repositories on Cloud article (#4162)
umbraco/UmbracoDocs: Umbraco-Cloud/Getting-Started/Repository-in-a-Cloud-Project/index.md

---
versionFrom: 7.0.0
versionTo: 10.0.0
---

# Repositories in an Umbraco cloud Project

> Each Umbraco cloud Project can have multiple environments: Development, Staging, and Live depending on your cloud project plan. Each environment has its own git repository that is hosted on Azure. Ideally, your Umbraco Cloud setup should look like this:

Reviewer: ```suggestion
Each Umbraco cloud Project can have multiple environments: Development, Staging and Live depending on your cloud project plan. Each environment has it's own git repository that is hosted on Azure. Ideally, your Umbraco Cloud setup should look like this: 
```

- [A source control repository with your own code](#a-source-control-repository-with-your-own-code)
- [A Umbraco Cloud source control repository with the locally cloned Umbraco project](#a-git-umbraco-cloud-source-control-repository-with-the-locally-cloned-umbraco-project)

:::note
Umbraco cloud repositories are only deployment repositories and should not be used as source code repositories.
:::

## A source control repository with your own code

> Source control is a way to control changes to files and directories. You can keep a record of changes and revert to specific versions of a file in the event you would like to back up to an earlier time. A source control repository is used as the single source of truth that has the latest version of your project source code with all the git branches.

Reviewer: ```suggestion
Source control is a way to control changes to files and directories. You can keep a record of changes and revert to specific versions of a file in the event you would like to back up to an earlier time. A source control repository is used as the single source of truth which has the latest version of your project source code with all the git branches.
```

There are different source code management tools that you can use such as GitHub, Git, GitLab, Apache Subversion (SVN), CVS, Mercurial, etc.

> You need to put your custom code in a different source control repository you prefer. You can verify that it works on your local machine and copy those changes to the Umbraco Cloud source control repository with the locally cloned Umbraco project. The locally cloned Umbraco project has a `.gitignore` file. This file ignores all the Visual Studio temporary files, build results, files generated by popular Visual Studio add-ons, Umbraco folder, media files, log files, Packages and Plugins, etc. while pushing the changes to your cloud environments.

Reviewer: ```suggestion
You need to put your custom code in a different source control repository of your choice. You can verify that it works on your local machine and copy those changes to the Umbraco Cloud source control repository with the locally cloned Umbraco project. The locally cloned Umbraco project has a `.gitignore` file. This file ignores all the Visual Studio temporary files, build results, files generated by popular Visual Studio add-ons, Umbraco folder, media files, log files, Packages and Plugins, etc. while pushing the changes to your cloud environments.
```
